source/threads.tex

```diff
@@ -12006,3 +12006,455 @@
 deadlock can occur.
 \end{note}
 \end{itemdescr}
+
+\rSec2[saferecl.hp]{Hazard pointers}
+
+\rSec3[saferecl.hp.general]{General}
+
+\pnum
+A hazard pointer is a single-writer multi-reader pointer
+that can be owned by at most one thread at any time.
+Only the owner of the hazard pointer can set its value,
+while any number of threads may read its value.
+The owner thread sets the value of a hazard pointer to point to an object
+in order to indicate to concurrent threads---which
+may delete such an object---that
+the object is not yet safe to delete.
+
+\pnum
+A class type \tcode{T} is \defn{hazard-protectable}
+if it has exactly one base class of type \tcode{hazard_pointer_obj_base<T,D>}
+for some \tcode{D},
+that base is public and non-virtual, and
+it has no base classes of type \tcode{hazard_pointer_obj_base<T2,D2>}
+for any other combination \tcode{T2}, \tcode{D2}.
+An object is \defn{hazard-protectable} if it is of hazard-protectable type.
+
+\pnum
+The time span between creation and destruction of a hazard pointer $h$
+is partitioned into a series of \defnadj{protection}{epochs};
+in each protection epoch,
+$h$ either is associated with a hazard-protectable object, or is unassociated.
```

Comment:
The paper styles "associated with" and "unassociated" in italics.

Reply:
That's true, but we don't always italicize local terms, e.g. see shared_ptr's "empty". I think this is fine.

Reply:
On the other hand, "owns" below is a definition, so we might as well make this one one, too.